### ISSUE TITLE
Timeseries for lizard catalogue

### ISSUE THREAD

**Issue by Derryrover** [nens/lizard-catalogue#111]
There is a new design for the catalog.

**Design link:**

https://xd.adobe.com/spec/c788afcd-664b-4c73-7a9d-7cdab52cc7a4-b1b9/

**How to use:**

- Switch between details and actions
- When in actions, click on select time series
- Click on the date picker to see how it expands
- Deselect Waterlevel in observation types
- Deselect Temperatuur in observation types
- Hover over the point in the middle of the map
- Click on the map, you will now have selected two points

**Notes:**

Since we are going to use a library to select a polygon, i just made a button for now.

The modal of selecting time series should be an overlay, now the screen shows a black background

**General improvements:**

- Better view of the map
- Better spacing
- Different approach with details/actions tabs
- Redesign of filtering
- Slightly different layout
- Time series
- Selection screen for time series

**API Endpoints needed:**

-

**Comment by Derryrover** [comment]
@joeriverheijden 
@lexvand 

Hi Joeri,

I made this ticket for the catalogue timeseries.
As soon as you know the exact requirements you can update this ticket.

kr Tom

**Comment by hoanphungt** [comment]
@lexvand @LirryPinter 
Hi guys,

Regarding the design, there is a button to open the selected timeseries in the Lizard Portal. However, the timeseries can only be viewed in the Lizard Portal via its related asset (for example this one: https://nxt3.staging.lizard.net/en/map/topography,assetgroup$b1d7f43/point/overflow$4799/@52.1037,5.0040,20/May,20,2020-Jun,01,2020). The timeseries catalogue does not show the asset so it's probably not possible to open the selected timeseries in the Portal.

Another alternative is to open the selected locations in the Portal. However, locations can also only be opened via its related asset.

Since this is the only issue left for the timeseries catalogue, I would probably open a new separate ticket for this.

I would like to hear your opinion/idea.

Kr, Hoan

**Comment by hoanphungt** [comment]
Hi @LirryPinter @lexvand,

When open multiple timeseries/locations in the API/Lizard portal, the first one is opened in a new tab. However, the 2nd, 3rd and so on are blocked by Chrome by default (similar to issue with the `Download All` button in the live site). This will only happen for the first time and if we unblock it, it will not happen again. 

However, I am thinking of having extra notice for users about this issue, please let me know what you think. Something like in this screenshot:
<img width="947" alt="Screenshot 2020-08-27 at 16 45 24" src="https://user-images.githubusercontent.com/43402556/91458102-6b7efc00-e885-11ea-876b-32781f654130.png">

**Comment by lexvand** [comment]
@hoanphungt You can select multiple assets at once in the Lizard Client, so no need to open multiple tabs.

Example: https://nxt3.staging.lizard.net/en/map/topography,assetgroup$b1d7f43/multi-point/overflow$4791+overflow$4793+overflow$4775/@52.0903,5.1181,16/-2Days0Hours+0Days3Hours
Or chart mode: https://nxt3.staging.lizard.net/en/charts/topography,assetgroup$b1d7f43/multi-point/overflow$4791+overflow$4793+overflow$4775/@52.0903,5.1181,16/-2Days0Hours+0Days3Hours

**Comment by hoanphungt** [comment]
@lexvand 
Great! we can multiple selections at once for the `Open in portal` button. However, still the `Open in API` button, I suspect that we would still need to open multiple tabs for multiple selections though. What do you think?

**Comment by lexvand** [comment]
Sorry, hadn't seen it also concerned the "Open in API" feature.

It doesn't seem user friendly to me to open a whole bunch of tabs. What if a selection consists of 10+ items? So either the API should support the filtering that leads to the selection (would require something like "timeseries__id in {id_list}") or we should support opening in API only for single timeseries.

@LirryPinter What's your opinion?

**Comment by LirryPinter** [comment]
@lexvand @hoanphungt 

These are valid points about the opening in the API. I thought that these filters were also on the the endpoint? Maybe @caspervdw knows this? 

Else I agree  with Lex that we can't open 10 tabs at the same time. So we need to find a solution for that.

**Comment by hoanphungt** [comment]
@LirryPinter @lexvand 
I think it would be nice to open them all at once. I tried to open multiple timeseries with a query like this but it didn't work: `/api/v4/timeseries/?timeseries__uuids=uuid1,uuid2,uuid3`. 

Otherwise, we can for now support open in API for only when user selects a single timeseries. If multiple timeseries are selected then we can disable the button and show a tooltip that `More than 1 time-series is selected`

**Comment by hoanphungt** [comment]
@lexvand @joeriverheijden @LirryPinter @Derryrover 
Hi Lex, Joeri, Tom and Lirry,

I just deployed the Time-series Catalogue to staging with the following changes:

- New design of the whole app
- New filter bar
- Catalogue for time-series with list of monitoring networks and their details
- Time-series selection modal with:

+ map of locations with search functionality
+ filter on observation types
+ list of selections on map
+ duration selection
+ Actions: open in API/Lizard and Export to .xlsx

Please let me know your feedbacks.

Kind regards,
Hoan

**Comment by Derryrover** [comment]
> @hoanphungt @Derryrover Hopefully this schema is helpful.
> ![Datamodel around timeseries (for catalogue)](https://user-images.githubusercontent.com/5370809/90895102-e8136580-e3c1-11ea-9e4a-de537b489bd7.png)

@lexvand
@caspervdw  
@joeriverheijden 

Hi Lex, Joeri and Casper,

Thanks still for the entity relation diagram above (made by Lex). 
It is very helpful.

Some things in the current entity relationship diagram (and thus in the Lizard database) strike me as odd.
Maybe because of my lack of understanding, or maybe cause of technical, compatibility or legacy reasons.
Maybe you could help me understand.
In my opinion the entity relations of any program acts as the "backbone" of the app. 
It needs to reflect the domain model and be clearly communicated across all teams.

My questions:
- Why are assets and timeseries linked via location? It would make more sense to me if a timeseries was linked to a asset and a asset to a location. If I am not wrong this is also causing severe hacks in the parramatta dashboards
- Why are timeseries directly part of monitoring network?
It would make more sense to me if assets were connected to the monitoring network instead.
- Also maybe "observation-types, nodes, data-sources, timeseries-types" could be better linked to the asset then to the timeseries itself.

Please let me know your thoughts

kr Tom

**Comment by lexvand** [comment]
My answers:

1. This is because of the different sources for the data in the systems of our customers. On one hand we've got asset management (source for assets, dûh) and telemetry (source of timeseries). In those telemetry systems there is often very little asset information, just a "location"/"station" with one or more "sensors". The Lizard model for timeseries has been largely based on that of FEWS, which also includes locations. We import the two data flows and try to link them through this relation between assets and locations. BTW: The SPOC (Single Point Of Configuration) systems we built for some of our customers also serves the purpose of combining the information of different sources.
We have looked at the possibility to get rid of locations as a separate model and treat it like an asset, but until now it has turned out to be too complicated.
2. One location can have multiple timeseries with different parameters, e.g. rain and water level measurements. We want to support networks for specific parameters, which requires a relation specifically to timeseries. Also, the monitoring networks should support timeseries that are not related to an asset.
3. These are actual timeseries properties. These properties can differ for the same asset, e.g. measurements versus model forecasted timeseries.
